keras_nlp/utils/text_generation.py

```diff
@@ -134,40 +125,63 @@ def token_probability_fn(inputs):
     ```
 
     """
-    prompt = validate_prompt(prompt)
+    prompt = _validate_prompt(prompt)
 
     input_is_1d = prompt.shape.rank == 1
     if input_is_1d:
         prompt = prompt[tf.newaxis, :]
-    validate_token_probability_fn(token_probability_fn, prompt)
 
-    shape = tf.shape(prompt)
-    batch_size = shape[0]
-    length = shape[1]
+    _validate_token_probability_fn(token_probability_fn, prompt)
+
+    # Maintain a mask tensor of shape `(batch_size,max_length)` which will
```

Comment:
Does every util need this verbatim? Could we move this into `_validate_prompt` or keep it is a standalone helper `_pad_prompt`?

Reply:
Hmmm, right. I reversed all forms of refactoring for this PR. Earlier, I'd put padding in `_generate_function`, which was a common point for all text gen functions. Should we keep this as it is for now, and wait for the PR where I refactor this file?

Edit: I've dumped these lines in a helper function for now.